script.js

```diff
@@ -69,14 +85,60 @@ function shuffleCards(array) {
 let shuffledCardsArray = shuffleCards(cardObjectsArray);
 let cardDivArray = document.querySelectorAll('.card');
 
-for(let i = 0; i < cardDivArray.length; i++) {
+for (let i = 0; i < cardDivArray.length; i++) {
     cardDivArray[i].innerHTML = '<img class="hamster" src="' + shuffledCardsArray[i].frontOfCard + '">'
-        + '<img class="sloth" src="images/cardBack.jpg">';
+        + '<img class="sloth" data-pair ="' + shuffledCardsArray[i].pairID + '" src= "images/cardBack.jpg">';
+
 }
 
+let clickCounter = 0
+let turnCounter = 0
+let firstClickPair = "start"
+let secondClickPair = "start"
 let slothClickArray = document.querySelectorAll('.sloth');
-slothClickArray.forEach(sloth => {
-    sloth.addEventListener('click', (e)=>{
-        sloth.classList.add('hide');
-    })
-})
+let matchedPairs = 0
+
+function addClickListener(onClick) {
+    slothClickArray.forEach(sloth => {
+            sloth.addEventListener('click', onClick )
+        })
+}
+
+function onClick() {
+    this.classList.add('hide');
+    this.classList.add('selected');
+    if (clickCounter  === 0) {
+        firstClickPair = this.dataset.pair
+        clickCounter++
+    } else {
+        clickCounter--
+        turnCounter ++
+        secondClickPair = this.dataset.pair
+        evaluateCards(onClick,slothClickArray)
+    }
+}
+
+    function flipCardsBack() {
+        let selectedCards = document.querySelectorAll('.selected');
+        selectedCards.forEach(card => {
+            card.classList.remove('hide')
+            card.classList.remove('selected')
+            addClickListener(onClick)
+        })
+    }
+
+    function evaluateCards(onClick, slothClickArray) {
```

Comment:
don't pass `onClick` as parameter - it will work anyway

Reply:
done - harry